omod/src/main/java/org/openmrs/module/htmlformentryui/page/controller/htmlform/ViewEncounterWithHtmlFormPageController.java

```diff
@@ -26,6 +28,19 @@ public void get(@RequestParam("encounter") Encounter encounter,
 
         patient.setPatient(encounter.getPatient());
 
+        AdministrationService administrationService = Context.getAdministrationService();
+        
+        String customPrintProvider = administrationService.getGlobalProperty("htmlformentryui.customPrintProvider");
+        String customPrintPageName = administrationService.getGlobalProperty("htmlformentryui.customPrintPageName");
+        String customPrintTarget = administrationService.getGlobalProperty("htmlformentryui.customPrintTarget");
+        if (StringUtils.isBlank(customPrintTarget)) {
+            customPrintTarget = "_self";
```

Comment:
I find it a little weird to see pure frontend stuff configured here, why not just set this as the default in **config.xml**?

Reply:
The browser default is _self, but I'm also passing target as a GET param explicitly, so one hand knows what the other hand is doing. (It's being passed to the server since opening a new tab is a client side thing, but responding to a request with a file download is a server side thing). In our specific use case, without it, it would open a new tab, but then download the file instead of displaying it.

Reply:
Making it the default gp value allows someone/thing to remove it. It's a design decision to "force" the use of the api in a certain way. The controller developer can choose to ignore it if they want, but should be provided the information in the case it is needed as a condition.

Reply:
change made in latest commit